core/metrics-core-service/src/test/java/org/hawkular/metrics/core/service/metrics/ExpressionTagQueryITest.java

```diff
@@ -0,0 +1,173 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.metrics.core.service.metrics;
+
+import static java.util.Arrays.asList;
+
+import static org.hawkular.metrics.model.MetricType.AVAILABILITY;
+import static org.hawkular.metrics.model.MetricType.COUNTER;
+import static org.hawkular.metrics.model.MetricType.GAUGE;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.hawkular.metrics.core.service.tags.ExpressionTagQueryParser;
+import org.hawkular.metrics.model.DataPoint;
+import org.hawkular.metrics.model.Metric;
+import org.hawkular.metrics.model.MetricId;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import rx.Observable;
+
+/**
+ *
+ * @author Stefan Negrea
+ */
+public class ExpressionTagQueryITest extends BaseMetricsITest {
+
+    @Test
+    public void tagValueSearch() throws Exception {
+        String tenantId = "jsonT1Tag";
+
+        createTagMetrics(tenantId);
+
+        ExpressionTagQueryParser test = new ExpressionTagQueryParser(dataAccess, metricsService);
+
+        List<Metric<Double>> gauges = test.parse(tenantId, GAUGE, "a1 ='1'").toList().toBlocking()
+                .lastOrDefault(null);
+        assertMetricListById(gauges, "m1");
+
+        gauges = test.parse(tenantId, GAUGE, "a1 != '1'").toList().toBlocking()
+                .lastOrDefault(null);
+        assertMetricListById(gauges, "m2", "m3", "m4", "m5");
+
+        gauges = test.parse(tenantId, GAUGE, "a1 ='1' OR a2='2'").toList().toBlocking()
+                .lastOrDefault(null);
+        assertMetricListById(gauges, "m1", "m6");
+
+        gauges = test.parse(tenantId, GAUGE, "a1 ='11' OR a2 = '22'").toList().toBlocking()
+                .lastOrDefault(null);
+        assertMetricListById(gauges);
+
+        gauges = test.parse(tenantId, GAUGE, "a1='2' AND (a2='3' OR a2='4')").toList().toBlocking()
+                .lastOrDefault(null);
+        assertMetricListById(gauges, "m4", "m5");
+
+        gauges = test.parse(tenantId, GAUGE, "a1 ='2' AND (a2 in ['3', '4'])").toList().toBlocking()
+                .lastOrDefault(null);
+        assertMetricListById(gauges, "m4", "m5");
+
+        gauges = test.parse(tenantId, GAUGE, "a1 ='2' AND (a2 notin ['3'])").toList().toBlocking()
+                .lastOrDefault(null);
+        assertMetricListById(gauges, "m5");
+
+        gauges = test.parse(tenantId, GAUGE, "hostname ='web.*'").toList().toBlocking()
+                .lastOrDefault(null);
+        assertMetricListById(gauges, "mA", "mB");
+
+        gauges = test.parse(tenantId, GAUGE, "hostname ='web.*' or a1='*'").toList().toBlocking()
```

Comment:
Minor thing but just in case it escaped into the review.

> a1 = '*'

The '*' is not a valid regexp in java, I guess you would mean '.*' to get all possible values, unless we want to shortcut this in a customized way.

I commented, because I saw it also in the first comment on the PR.